qbft/spectest/tests/commit/invalid_full_data.go

```diff
@@ -1,8 +1,34 @@
 package commit
 
-import "github.com/bloxapp/ssv-spec/qbft/spectest/tests"
+import (
+	"github.com/bloxapp/ssv-spec/qbft"
+	"github.com/bloxapp/ssv-spec/qbft/spectest/tests"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+)
 
 // InvalidFullData tests signed commit with an invalid full data field (H(full data) != root)
 func InvalidFullData() tests.SpecTest {
-	panic("implement")
+	pre := testingutils.BaseInstance()
+	ks := testingutils.Testing4SharesSet()
+
+	msgs := []*qbft.SignedMessage{
+		testingutils.TestingProposalMessage(ks.Shares[1], 1),
+
+		testingutils.TestingPrepareMessage(ks.Shares[1], 1),
+		testingutils.TestingPrepareMessage(ks.Shares[2], 2),
+		testingutils.TestingPrepareMessage(ks.Shares[3], 3),
+
+		testingutils.TestingCommitMessageWithFullData(ks.Shares[1], 1, testingutils.DifferentFullData),
+		testingutils.TestingCommitMessageWithFullData(ks.Shares[2], 2, testingutils.DifferentFullData),
+		testingutils.TestingCommitMessageWithFullData(ks.Shares[3], 3, testingutils.DifferentFullData),
+	}
+	return &tests.MsgProcessingSpecTest{
+		Name:          "commit with invalid full data",
+		Pre:           pre,
+		InputMessages: msgs,
+		OutputMessages: []*qbft.SignedMessage{
+			testingutils.TestingPrepareMessage(ks.Shares[1], 1),
+			testingutils.TestingCommitMessage(ks.Shares[1], 1),
+		},
+	}
 }
```

Comment:
Yes, I agree 100% with your observation.
I talked about it in the PR description. May you take a look at it?
Since we don't use the `FullData` field when receiving the commit message, there shouldn't be a problem related to the QBFT execution.
However, regarding security, one can add a `FullData` field with maximum length to slow down the node processing.
For that, we can add a check that asserts that the field is empty in the commit message. What do you think?

Reply:
opened #329 
add a comment in code that refers to this issue please